sycl-fusion/jit-compiler/include/KernelFusion.h

```diff
@@ -9,58 +9,68 @@
 #ifndef SYCL_FUSION_JIT_COMPILER_KERNELFUSION_H
 #define SYCL_FUSION_JIT_COMPILER_KERNELFUSION_H
 
+#include "DynArray.h"
 #include "Kernel.h"
 #include "Options.h"
 #include "Parameter.h"
+
 #include <cassert>
-#include <string>
-#include <variant>
-#include <vector>
 
 namespace jit_compiler {
 
 class FusionResult {
 public:
-  explicit FusionResult(std::string &&ErrorMessage)
-      : Type{FusionResultType::FAILED}, Value{std::move(ErrorMessage)} {}
+  explicit FusionResult(const char *ErrorMessage)
+      : Type{FusionResultType::FAILED}, KernelInfo{ErrorMessage, 0} {}
```

Comment:
I reckon that storing the error message in the otherwise unused `KernelInfo` object to replace the use of `std::variant` might be thought just a tad too much around the corner... WDYT?

Reply:
We can have our class being a [tagged union](https://en.wikipedia.org/wiki/Tagged_union):
```
union { 
  DynString ErrorMessage;
  SYCLKernelInfo KernelInfo;
} Value;
```
The `Type` member would be the tag used for type safety. This would be the no-STL version of what we did before.

Reply:
I like the union idea, however that gets complicated because of the member types define non-trivial special member functions and constructors, so most implicitly defined members of the union are deleted. I didn't flesh that approach out completely. In the end, I think just spending the few extra bytes for having a separate `DynString` is the cleanest way to go.

Reply:
Fine, we can live with this compromise I suppose :+1: